### PR TITLE
Workerfixes

### PR DESCRIPTION
**Pull request by nathanielknight**
This branch contains fixes necessary to get Kive tasks running on the worker node.

If it would be helpful, I could write up these requirements, but the code is fairly explicit.

### CODE REVIEW

File: vagrant/Vagrantfile [hunk 1]
```diff
@@ -13,12 +13,16 @@
...
+Dir.mkdir("./data") if not Dir.exists?("./data")
+
 Vagrant.configure("2") do |config|
 
   config.vm.box = "geerlingguy/centos7"
   config.vm.box_version = "1.2.6"
 
   config.vm.synced_folder "..", "/usr/local/share/Kive"
+  config.vm.synced_folder "./data", "/data",
+    mount_options: ["dmode=777", "fmode=777"]
```
Comment (donkirkby) [inline vagrant/Vagrantfile:25]: Does that fmode make sense, or would 666 be better? I'm not familiar with the option, so I'm just guessing how it works. For Kive in particular, I can't think of any reason why we'd need executable files under the media root.

File: vagrant/envvars.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# This section gets appended to the kive user's .bash_profile
+# and to /etc/sysconfig/httpd
+export KIVE_DB_NAME=kive
+export KIVE_DB_USER=kive
+export KIVE_DB_HOST=head
+export KIVE_DB_PASSWORD=YZcGRH8AZAj6VAF6qj8KOy
```
Comment (donkirkby) [inline vagrant/envvars.conf:6]: If we're going to include a password in source control, we should probably have a comment explaining why. One strategy I've used in the past is to generate a random password during dev/test setup and write it into a configuration file.

File: vagrant/setuplib.sh [hunk 3]
```diff
@@ -214,20 +247,58 @@ function setuplib::python3 {
...
+    echo "========= Installing Kive worker components ==========="
+    usermod -a -G kive vagrant
+    yum groupinstall -q -y 'Development Tools'
+    yum install -q -y python-devel libsqlite3x-devel words lsof graphviz graphviz-devel
+
+    python3 -m venv /opt/venv_kive
+    . /opt/venv_kive/bin/activate
+    python -m pip install --upgrade pip
+    pushd /usr/local/share/Kive/
+    python -m pip install -r requirements-dev.txt
```
Comment (donkirkby) [inline vagrant/setuplib.sh:262]: Is `requirements-dev.txt` really needed? I would guess that `requirements.txt` would be enough.

File: vagrant/setuplib.sh [hunk 4]
```diff
@@ -214,20 +247,58 @@ function setuplib::python3 {
...
 }
 
-function setuplib::kive {
-    echo "========= Installing Kive ==========="
+# Set a node up to host the Kive web application
+function setuplib::kive_head {
+    echo "========= Installing Kive application ==========="
     python3 -m venv /opt/venv_kive
     cd /usr/local/share/Kive/vagrant
     ./kive_setup.bash requirements-dev.txt
```
Comment (donkirkby) [inline vagrant/setuplib.sh:271]: Same question here about the requirements file, although the dev version might actually be useful for doing development work on the head node. However, I suggest we install `requirements.txt` by default, and then manually install the rest if needed.

File: vagrant/setuplib.sh [hunk 5]
```diff
@@ -214,20 +247,58 @@ function setuplib::python3 {
...
-    cd /usr/local/share/Kive/vagrant_ubuntu
-    ./dbcreate.sh
-}
\ No newline at end of file
+
+    local secretkey
+    secretkey="$(python3 -c 'import secrets; print(secrets.token_urlsafe())')"
```
Comment (donkirkby) [inline vagrant/setuplib.sh:274]: Please add a comment explaining what's happening here.

File: vagrant/Vagrantfile (hunk 1 above)
Comment (nathanielknight) [reply vagrant/Vagrantfile:25]: As I understand it, the directory needs to be "executable" so that other users can enter and modify it.

Comment (donkirkby) [reply vagrant/Vagrantfile:25]: Maybe I'm misunderstanding what `fmode` and `dmode` are used for. I assumed that `fmode` would be applied to files and `dmode` to directories. Also, are they just replacing the existing file attributes or are they masking them? If they're used as a mask, then just add an explanatory comment and leave them as is.

**Pull request merged**